source/develop/infra/testing/lago/testing-engine-patches-with-lago.html.md

```diff
@@ -0,0 +1,74 @@
+
+## Creating the test suite
+
+ovirt-system-tests is composed by several test suites designed for CI testing. To test our patches, we will copy one of these test-suites and change it to suit our needs.
+
+The first thing to do is select which test suite can be used as base for our manual test. This guide will be testing a patch for ovirt-engine's master branch, then it will be based on `basic_suite_master`.
+
+Inside the ovirt-system-tests repository, copy the test suite directory:
+
+    $ cp -r basic_suite_master manual_suite_master
+
+Now we need to edit the Lago initialization file. This file is usually a symlink to a common initialization file. We will copy the actual file, so we can change it.
+
+    $ cd manual_suite_master
+    $ cp --remove-destination $(realpath LagoInitFile.in) LagoInitFile.in
+
+Edit the `LagoInitFile.in` file to suit your needs. Easier way to do it is keep the predefined machines and just edit their parameters, if they are reasonable for your tests. If you need more machines to reproduce a real environment, or less machines to save resources, you'll need to edit the test scenarios later.
+
+## Editing test scenarios
+
+ovirt-system-tests' suites come with test scenarios, that are ran by Lago. As the test suite we are using as base was designed to test ovirt-engine setup, these steps are part of test scenarios, not of deployment scripts. We will still need to run these tests to get engine up and running.
+
+We need to remove the tests that are not doing engine setup. Please note that, depending on the patch you are testing, you may want to keep those tests/update them to test the new behaviour introduced by your patch):
+
+    $ cd test-scenarios
+    $ rm 004_basic_sanity.py 005_aaa-ldap.py
+
+Also, if you added or removed machines in the Lago initialization file, you'll need to edit `002_bootstrap.py` to reflect the new environment, implementing the code needed to add the new machines to the engine. You may also want to check the engine-setup answer file used by `001_initialize_engine.py` and modify it if you need some different answers.
+
+## Preparing a build with your patch
+
+To be able to use your modifications in lago, you must build a modified RPM. Lago is built on [repoman](http://repoman.readthedocs.io/), so it can fetch RPMs from several places, including [oVirt Jenkins instance](http://jenkins.ovirt.org/).
+
+The easier way to build an RPM is using jenkins' "build-artifacts" jobs, but they take around 1 hour to run, and the artifacts will be removed after some time. For this example we will use the following job: [http://jenkins.ovirt.org/job/ovirt-engine\_master\_build-artifacts-el7-x86\_64/](http://jenkins.ovirt.org/job/ovirt-engine_master_build-artifacts-el7-x86_64/).
+
```

Comment:
I suggest to use build-manual jobs instead of build-artifacts jobs. build-artifatcs jobs are used also by publisher jobs and the temporary build may be grabbed and published on snapshot repos.